EpsilonDriverControls/Src/DriverControls.c

```diff
@@ -256,13 +256,17 @@ void sendDriveCommandsTask(void const* arg)
         {
             HAL_GPIO_TogglePin(LED_BLUE_GPIO_Port, LED_BLUE_Pin);
             motorVelocityOut = MAX_FORWARD_RPM;
-            motorCurrentOut = calculateMotorCurrent(accelPercentage);
+
+            // Smoothing output current so that there aren't any big jumps that could cause the motors to trip. Essentially a simple IIR low pass filter
+            motorCurrentOut += MOTOR_CURRENT_SMOOTHING_FACTOR * (calculateMotorCurrent(accelPercentage) - motorCurrentOut);
```

Comment:
I think the logic of smoothing the current should be put into `calculcateMotorCurrent`, so that this function just has to call:
`motorCurrentOut = calculateMotorCurrent(motorCurrentOut)`

Reply:
This is a bit more complicated with the regen calculation because the regen scale factor is applied to the return value of calculateMotorCurrent, then the smoothing is done

Reply:
This is a bit more complicated with regen because the regen scaling is applied to the return value of `calculateMotorCurrent` then the smoothing is done. Would it be alright if I created two functions called `calculateAccelMotorCurrent` and `calculateRegenMotorCurrent`, then make those functions call `calculateMotorCurrent` as it is, then have the smoothing done in `calculateAccelMotorCurrent` and `calculateRegenMotorCurrent` with the different functionality needed in each?

Reply:
One more thing, when the percentage is less than the non-zero threshold, `calculateMotorCurrent` returns 0, but with the smoothing, it wouldn't instantly go to 0, but would slowly go to 0 if it was at a non-zero value before. Is this okay, or should I add the functionality so that if 0 is returned, then smoothing is skipped and 0 is assigned?

Reply:
That sounds good! As for the returning 0, it's OK if it smooths over.